garak/attempt.py

```diff
@@ -42,38 +68,24 @@ def __init__(
         probe_classname=None,
         probe_params=None,
         targets=None,
-        outputs=None,
         notes=None,
         detector_results=None,
         goal=None,
         seq=-1,
     ) -> None:
         self.uuid = uuid.uuid4()
+        self.messages = []
```

Comment:
Are we happy with a list of dicts? Does it make sense to implement an abstract base class for this?

Reply:
That way we can enforce roles, ordering, etc.